scipy/special/cephes/hyp2f1.c

```diff
@@ -130,7 +167,15 @@ double a, b, c, x;
     if (ax < 1.0 || x == -1.0) {
 	/* 2F1(a,b;b;x) = (1-x)**(-a) */
 	if (fabs(b - c) < EPS) {	/* b = c */
-	    y = pow(s, -a);	/* s to the -a power */
+	    if (b < 0 && b > -100 && (floor(b) - b) < EPS) {
+		/*
+		    The approximation becomes good enough 
+		    again once we go below -100
```

Comment:
It's not an approximation --- the formula is exact. The point is that when c is integer <= 0, the result is not uniquely defined by continuity.

Also, I think the statement is not correct? For example, if `x > 1`, the equation below produces nan but the sum still terminates.

Reply:
`if ax < 1.0`